tests/unit/valgrind.suppressions

```diff
@@ -10,7 +13,51 @@
    fun:pthread_create@@GLIBC_2.2.5
    fun:main
 }
-
+# This block is a similar suppression to the pthread suppressions above, but for a different libc version.
+# These are the suppressions for pthread_create in GLIBC_2.34.
+# Remove the previous pthread suppression block after the new CTest memcheck is fully integrated.
+# Tracking issue: https://github.com/aws/s2n-tls/issues/4777
+{
+   pthread_false_positive_ubuntu22
```

Comment:
I actually think that if we make the suppression name a bit more descriptive then some of the comments might not be necessary. 
```suggestion
   pthread_false_positive_glibc_2_34
```

Reply:
I only remove those comments where it states that these two suppressions is for GLIBC false positive. I think other comments should remain there, since they are helpful to understand the purpose and context of such suppressions.